### PR TITLE
Preparation for next PR: move code and add tests

### PR DESCRIPTION
**Pull request by thijs-obj**
Changes:
1) Added some tests
2) Moved some code

This is a preparation for PR https://github.com/objectiv/objectiv-analytics/pull/406

@KathiaBarahona @jansentom please have a look

### CODE REVIEW

File: bach/sql_models/graph_operations.py [hunk 1]
```diff
@@ -165,35 +165,6 @@ def find_node(
     return result[0]
 
 
-def replace_node_in_graph(start_node: SqlModel,
```
Comment (thijs-obj) [inline bach/sql_models/graph_operations.py:168]: Moved this lower in the file, so it's grouped with the helper function (`_replace_model_in_graph_recursively`) that it calls.

Comment (thijs-obj) [inline bach/sql_models/graph_operations.py:168]: Moved this lower in the file, so it's grouped with the helper function (`_replace_model_in_graph_recursively`) that it calls.

**Review by jansentom — APPROVED** [review]
LGTM, nice new tests.

**Review by KathiaBarahona — APPROVED** [review]
lgtm

**Pull request merged**